shm/coordinates.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+
+class Coordinates {
```

Comment:
This is good candidate for constexpr class :)

Reply:
Please try to make it constexpr. This will be a good exercise and not too hard :)